sdk/core/azure-core/src/main/java/com/azure/core/util/Context.java

```diff
@@ -256,15 +265,21 @@ public Optional<Object> getData(Object key) {
      * @return A map containing all values of the context linked-list.
      */
     public Map<Object, Object> getValues() {
-        return getValuesHelper(new HashMap<>());
-    }
+        Map<Object, Object> map = new HashMap<>((int) Math.ceil(contextCount / 0.75F));
 
-    private Map<Object, Object> getValuesHelper(Map<Object, Object> values) {
-        if (key != null) {
-            values.putIfAbsent(key, value);
+        for (Context pointer = this; pointer != null; pointer = pointer.parent) {
+            if (pointer.key != null) {
+                map.putIfAbsent(pointer.key, pointer.value);
+            }
+
+            // If the contextCount is 1 that means the next parent Context is the NONE Context.
+            // Break out of the loop to prevent a meaningless check.
+            if (pointer.contextCount == 1) {
+                break;
+            }
```

Comment:
It's unlikely to be better, but has any consideration been given to a virtualized Map here that implements AbstractMap and acts as a facade over Context?

Reply:
Hmm, I think it could be better where the design can become a shared `Map` for the specific `Context` instance, given `Context` is immutable or at least the chain of `Context`s are immutable. This shared `Map` can be immutable externally but mutable by the owning `Context` so that `get` calls will either have the value in the `Map` or check the `Context` chain and update the `Map` or indicate somehow that value doesn't exist. The `Map` iterators would need to eager populate the `Map`.

All that said, I think we're better off deprecating this method as its usage is something in the past before we added the better internal APIs for converting from our Context to Reactor's Context.